src/simple_model.jl

```diff
@@ -0,0 +1,153 @@
+export SimpleNLPModel, obj, grad, grad!, cons, cons!, jac_coord, jac, jprod,
+       jprod!, jtprod, jtprod!, hess, hprod, hprod!
+
+type SimpleNLPModel <: AbstractNLPModel
+  meta :: NLPModelMeta
+
+  counters :: Counters
+
+  # Functions
+  f :: Function
+  g :: Function
+  g! :: Function
+  H :: Function
+  c :: Function
+  c! :: Function
+  J :: Function
+end
+
+function SimpleNLPModel(x0; y0 = [], lvar = [], uvar = [], lcon = [], ucon = [],
+    nnzj = 0, nnzh = 0, obj = error, grad = error, grad! = error, hess =
```

Comment:
At least `obj` should be mandatory I believe (there's no problem without an objective). The others should probably be `Function`s, and if not specified, they should probably be something like `() -> throw(error("not implemented"))`. So the arguments to the constructor should probably be of type `Union{Nothing, Function}`, and if `Nothing`, then we should set them to the default value. What do you think?